rust/processor/src/utils/database.rs

```diff
@@ -206,26 +189,79 @@ where
     };
     let debug_string = diesel::debug_query::<diesel::pg::Pg, _>(&final_query).to_string();
     tracing::debug!("Executing query: {:?}", debug_string);
-
     let conn = &mut pool.get().await.map_err(|e| {
         tracing::warn!("Error getting connection from pool: {:?}", e);
         diesel::result::Error::DatabaseError(
             diesel::result::DatabaseErrorKind::UnableToSendCommand,
             Box::new(e.to_string()),
         )
     })?;
+    let res = final_query.execute(conn).await;
+    if let Err(ref e) = res {
+        tracing::warn!("Error running query: {:?}\n{:?}", e, debug_string);
+    }
+    res
+}
 
+pub async fn execute_with_better_error_conn<U>(
```

Comment:
Why do we need both execute_with_better_error_conn and execute_with_better_error?

Reply:
one uses the regular diesel, one uses diesel async; diesel async doesn't support `run_without_transaction` or whatever for migrations 😢